segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -470,7 +470,17 @@ public void mergeSegments(MergeSegments mergeSegments) {
         }
 
         log.info(mergeSegments.getRequestId(), "Merging Segments {} ", mergeSegments);
-        segmentStore.mergeStreamSegment(mergeSegments.getTarget(), mergeSegments.getSource(), TIMEOUT)
+
+        // Populate the AttributeUpdates for this mergeSegments operation, if any.
+        AttributeUpdateCollection attributeUpdates = new AttributeUpdateCollection();
+        if (mergeSegments.getAttributeUpdates() != null) {
+            for (WireCommands.ConditionalAttributeUpdate update : mergeSegments.getAttributeUpdates()) {
+                attributeUpdates.add(new AttributeUpdate(AttributeId.fromUUID(update.getAttributeId()),
+                    AttributeUpdateType.get(update.getAttributeUpdateType()), update.getNewValue(), update.getOldValue()));
```

Comment:
Thanks for adding `AttributeUpdateType`. One thing to do is we need to make sure that the values within `WireCommands.ConditionalAttributeUpdate` match those from `AttributeUpdateType`. See my comment in `WireCommands`.

Reply:
Done, added `REPLACE` and `REPLACE_IF_EQUALS` in `ConditionalAttributeUpdate`.